### ISSUE TITLE
Dialog for open-rewrite recipes shows "Loading..." and nothing happens

### ISSUE THREAD

**Issue by ahoehma** [spring-projects/sts4#1246]
**Describe the bug**
I try to select and run a recipe but there no recipes shown.

![image](https://github.com/spring-projects/sts4/assets/548214/cd1b6e03-3359-44ab-8bb2-374f5a1bbfa5)

I have these recipe filters:

![image](https://github.com/spring-projects/sts4/assets/548214/338d9a1b-1284-49d1-9533-94351f6d5b44)

**To Reproduce**

1. Right click on a Java File
2. Spring->Refactor....

**Versions**

Spring Tool Suite 4 

Version: 4.22.0.RELEASE
Build Id: 202403071427
Revision: 44291b464093d161f115d8e16059e5272ed3c81d

Copyright (c) 2007 - 2024 Broadcom
All rights reserved. Visit https://spring.io/tools

This product includes software developed by the
Eclipse Foundation https://www.eclipse.org

This product includes software developed by the
Apache Software Foundation https://www.apache.org

Lombok v1.18.32 "Envious Ferret" is installed. https://projectlombok.org/

**Comment by martinlippert** [comment]
There is a problem in that version with the dialog showing up twice and one of the dialogs stays in the "loading..." state. Do you see the same? Multiple dialogs showing up?

In addition to that it would be interesting to see what happens on the log output of the Spring Boot language server process. To get that log output, you would need to enable this via `Preferences -> Language Servers -> Spring Boot Language Server` and then enable the `Logging to Console` checkbox. After a restart, you will see log output being streamed to a console view. You should see something like

```
INFO  o.s.i.v.b.j.r.RewriteRecipeRepository - Loading Rewrite Recipes...
INFO  o.s.i.v.b.j.r.RewriteRecipeRepository - Done loading Rewrite Recipes
```

when you select that action. But maybe there is an error showing up, so would be interesting to see the output.

One side note: the refactoring recipes that this action offers run on the entire project, not just a single Java source file - just in case you are trying to use this specifically for one source file.

**Comment by ahoehma** [comment]
I installed the nightly version and now I see many recipes :D

![image](https://github.com/spring-projects/sts4/assets/548214/aa40b2f0-ab97-4586-92bf-4d32a0f6ac33)

**Comment by martinlippert** [comment]
@BoykoAlex fixed this with https://github.com/spring-projects/sts4/commit/71bcc7a580385ea1ae7aa7af1ae8bef30aaea50a, included in the `4.22.1` release already.

**Comment by martinlippert** [comment]
And thanks @ahoehma for checking this out and letting us know, much appreciated.
